code/game/objects/items/weapons/RCD.dm

```diff
@@ -382,3 +389,285 @@ RCD
 	name = "admin RCD"
 	max_matter = INFINITY
 	matter = INFINITY
+
+
+/obj/item/weapon/rcd/arcd
+	name = "advanced rapid-construction-device (ARCD)"
+	max_matter = 300
+	matter = 300
+	color = "red"
+
+
+// Ranged RCD
+
+
+/obj/item/weapon/rcd/arcd/afterattack(atom/A, mob/user, proximity)
+	proximity = 1
```

Comment:
couldn't I just put this in the args? I'm not exactly sure what youre suggesting, all the "camera RCD's" use the proximity override too so I need to be 100% sure before I go changing it.